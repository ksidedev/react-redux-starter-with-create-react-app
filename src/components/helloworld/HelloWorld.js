import React from 'react';
import PropTypes from 'prop-types';

const HelloWorld = (props) => {
  const { helloWorldText, onSubmit, className } = props;
  return (
    <div className={className}>
      <p onClick={onSubmit}>
        Click me
      </p>
      <p>{helloWorldText}</p>
    </div>
  );
};

HelloWorld.defaultProps = {
  helloWorldText: '',
  onSubmit: () => {},
  className: '',
};

HelloWorld.propTypes = {
  helloWorldText: PropTypes.string,
  onSubmit: PropTypes.func,
  className: PropTypes.string,
};

export default HelloWorld;
