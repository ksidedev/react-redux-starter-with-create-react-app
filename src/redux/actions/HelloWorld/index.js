export const helloWorldDefaultText = text => ({
  type: 'HELLO_WORLD_DEFAULT_TEXT',
  text,
});

export default helloWorldDefaultText;
