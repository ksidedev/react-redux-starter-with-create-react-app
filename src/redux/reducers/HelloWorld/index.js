export const initialState = {
  helloWorldDefaultText: 'Hello, World!',
};

export default (state = initialState, action) => {
  switch (action.type) {
    case 'HELLO_WORLD_DEFAULT_TEXT':
      return Object.assign({}, state, {
        helloWorldDefaultText: action.text,
      });
    default:
      return state;
  }
};
