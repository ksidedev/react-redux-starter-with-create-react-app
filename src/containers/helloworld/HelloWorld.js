import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { helloWorldDefaultText } from '../../redux/actions/HelloWorld';
import { HelloWorldStyled } from '../../components/helloworld/styles/js/styled';

class HelloWorld extends Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(event) {
    event.preventDefault();
    const { dispatch } = this.props;
    dispatch(helloWorldDefaultText('Text changes with redux successfully.'));
  }

  render() {
    const { helloWorldProps } = this.props;
    return (
      <>
        <HelloWorldStyled
          className="style-override-example"
          onSubmit={this.onSubmit}
        />
        <p>{helloWorldProps}</p>
      </>
    );
  }
}

HelloWorld.defaultProps = {
  dispatch: () => {},
  helloWorldProps: '',
};

HelloWorld.propTypes = {
  dispatch: PropTypes.func,
  helloWorldProps: PropTypes.string,
};

const HelloWorldToProps = state => ({
  helloWorldProps: state.HelloWorld.helloWorldDefaultText,
});

export default connect(HelloWorldToProps, undefined)(HelloWorld);
