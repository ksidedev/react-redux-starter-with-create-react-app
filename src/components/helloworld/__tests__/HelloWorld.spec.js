import React from 'react';
import renderer from 'react-test-renderer';
import HelloWorld from '../HelloWorldComponent';

test('it works', () => {
  const tree = renderer.create(<HelloWorld />).toJSON();
  expect(tree).toMatchSnapshot();
});
