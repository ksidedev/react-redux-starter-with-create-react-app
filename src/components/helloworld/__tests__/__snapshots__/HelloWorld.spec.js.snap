// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
<div
  className=""
>
  <p
    onClick={[Function]}
  >
    Click me
  </p>
  <p>
    
  </p>
</div>
`;
